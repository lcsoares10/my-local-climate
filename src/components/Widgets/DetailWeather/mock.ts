

export const mockDetailWheater = {
  clouds: 10,
  pop: 5,
  wind_speed: 100,
  humidity: 0,
  uvi: 70,
};
