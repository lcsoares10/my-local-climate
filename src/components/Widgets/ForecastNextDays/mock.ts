
export default [
    {
        "dt": "20/01/2022",
        "sunrise": 1642666992,
        "sunset": 1642714942,
        "moonrise": 1642722720,
        "moonset": 1642674720,
        "moon_phase": 0.58,
        "temp": {
            "day": 26.22,
            "min": 23.24,
            "max": 28.83,
            "night": 23.91,
            "eve": 28.23,
            "morn": 23.32
        },
        "feels_like": {
            "day": 26.22,
            "night": 24.73,
            "eve": 32.19,
            "morn": 24.11
        },
        "pressure": 1015,
        "humidity": 79,
        "dew_point": 21.94,
        "wind_speed": 5.51,
        "wind_deg": 107,
        "wind_gust": 7.92,
        "weather": [
            {
                "id": 800,
                "main": "Clear",
                "description": "c√©u limpo",
                "icon": "01d"
            }
        ],
        "clouds": 0,
        "pop": 0.46,
        "uvi": 14.65
    },
    {
        "dt": "21/01/2022",
        "sunrise": 1642753435,
        "sunset": 1642801333,
        "moonrise": 1642811220,
        "moonset": 1642764360,
        "moon_phase": 0.62,
        "temp": {
            "day": 25.94,
            "min": 22.89,
            "max": 26.15,
            "night": 23.41,
            "eve": 25.18,
            "morn": 23.04
        },
        "feels_like": {
            "day": 26.68,
            "night": 24.13,
            "eve": 25.87,
            "morn": 23.83
        },
        "pressure": 1014,
        "humidity": 80,
        "dew_point": 22.01,
        "wind_speed": 6.65,
        "wind_deg": 107,
        "wind_gust": 9.47,
        "weather": [
            {
                "id": 802,
                "main": "Clouds",
                "description": "nuvens dispersas",
                "icon": "03d"
            }
        ],
        "clouds": 31,
        "pop": 0.27,
        "uvi": 14.29
    },
 
];
