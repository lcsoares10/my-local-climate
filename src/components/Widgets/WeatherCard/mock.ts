export const weatherMock = {
  description: "Chuva leve",
  icon: "",
  id: 5,
  main: "chuva",
};

export const tempMock = 25;

export const dateCurrentMock = "21/01/2022 15:03";
export const dtMock = "21/01/2022"