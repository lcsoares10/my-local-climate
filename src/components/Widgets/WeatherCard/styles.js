import styled from "styled-components";

export const WheaterCard = styled.div`
  cursor: pointer;
  display: flex;
  flex-direction: column;
  align-items: center;
  background: rgba(255, 255, 255, 0.08);
  border: 1px solid #7a7a7a;
  box-sizing: border-box;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  border-radius: 20px;
  padding: 20px 30px;
  margin-left: 2rem;
`;
export const Description = styled.div`
  color: ${({ theme }) => theme.colors.white};
  font-weight: 300;
  margin-top: -15px;
`;

export const IconCard = styled.div`
  width: 100%;
  max-width: 120px;
`;

export const TemperatureCard = styled.div`
  font-weight: 300;
  font-size: 2rem;
  color: ${({ theme }) => theme.colors.white};
`;
