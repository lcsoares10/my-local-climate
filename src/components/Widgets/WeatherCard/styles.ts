import styled from "styled-components";
import { WeatherCardProps } from "./index";

export const WeatherCard = styled.div<WeatherCardProps>`
  cursor: ${({ inline }) => !inline && "pointer"};
  display: flex;
  gap: ${({ inline }) => inline && "2rem"};
  flex-direction: ${({ inline }) => (inline ? "row" : "column")};
  align-items: center;
  background: ${({ inline }) => !inline && "rgba(255, 255, 255, 0.08)"};
  border: ${({ inline }) => !inline && "1px solid #7a7a7a"};
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  border-radius: 20px;
  padding: 20px 30px;
  margin-left: ${({ inline }) => !inline && "2rem"};
  transition: 1s;
  flex-wrap:wrap &:hover {
    transform: ${({ hoverEffect }) => hoverEffect && "scale(1.04)"};
    transition: ${({ hoverEffect }) => hoverEffect && "0.5s"};
    background: ${({ hoverEffect }) => hoverEffect && "rgba(0, 0, 0, 0.5)"};
  }
`;
export const Description = styled.div<WeatherCardProps>`
  color: ${({ theme }) => theme.colors.white};
  font-weight: ${({ inline }) => inline && 600};
  margin-top: -15px;
  order: ${({ inline }) => inline && 2};
  font-size: ${({ inline }) => (inline ? "4rem" : "1rem")};
  text-transform: capitalize;
  @media (max-width: ${({ theme }) => theme.media.tablet}) {
    font-size: 3rem !important;
  }
  @media (max-width: ${({ theme }) => theme.media.mobile}) {
    font-size: 2rem !important;
  }
`;

export const IconCard = styled.div<WeatherCardProps>`
  width: 100%;
  max-width: ${({ inline }) => (inline ? "200px" : "120px")};
  order: ${({ inline }) => inline && 3};
  min-width: 100px;
`;

export const TemperatureCard = styled.div<WeatherCardProps>`
  font-weight: 300;
  font-size: ${({ inline }) => (inline ? "10rem" : "2rem")};
  color: ${({ theme }) => theme.colors.white};
  order: ${({ inline }) => inline && 1};
  @media (max-width: ${({ theme }) => theme.media.tablet}) {
    font-size: 8rem !important;
  }
  @media (max-width: ${({ theme }) => theme.media.mobile}) {
    font-size: 5rem !important;
  }
`;
