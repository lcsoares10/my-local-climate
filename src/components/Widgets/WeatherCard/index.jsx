import React from "react";
import * as S from "./styles";
import rain from "../../../../public/rain.gif";
import Image from "next/image";

function WeatherCard() {
  return (
    <S.WheaterCard>
      <S.TemperatureCard>24</S.TemperatureCard>
      <S.IconCard>
        <Image src={rain} alt="Picture of the author" />
      </S.IconCard>
      <S.Description>chuva moderada</S.Description>
    </S.WheaterCard>
  );
}

export default WeatherCard;
