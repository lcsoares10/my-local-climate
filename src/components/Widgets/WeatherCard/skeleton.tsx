import React from "react"
import ContentLoader from "react-content-loader"
import * as S from "./styles";

export const Skeleton = (props) => (
  <ContentLoader 
    speed={2}
    width="100%"
    viewBox="0 0 400 50"
    backgroundColor="#f3f3f334"
    foregroundColor="#ecebeb55"
    {...props}
  >
    <rect x="" y="0" rx="20" ry="20" width="100%" height="50" /> 


  </ContentLoader>
)

export const SkeletonWheaterCard = (props) => (

    <S.WeatherCard {...props}>
      <S.TemperatureCard {...props}>
       
      </S.TemperatureCard>
      <S.Description {...props}>
        <Skeleton></Skeleton>
      </S.Description>
      <S.Description {...props}>
        <Skeleton></Skeleton>
      </S.Description>
    </S.WeatherCard>

)

