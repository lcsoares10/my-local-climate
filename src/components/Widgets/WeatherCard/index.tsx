import React, { useEffect, useState } from "react";

import { Player } from "@lottiefiles/react-lottie-player";
import { getWeatherImage } from "../../../utils/listWeatherConditions";

import * as S from "./styles";
import { TWeather } from "../../../types/wheatherData";
import { SkeletonWheaterCard } from "./skeleton";

export type WeatherCardProps = {
  weather: TWeather;
  temp?: number;
  hoverEffect?: boolean;
  inline?: boolean;
  dt?: String;
  selected?: boolean;
  dateCurrent?: String;
};

const WeatherCard = ({ ...props }: WeatherCardProps) => {
  const [gifWeatherImage, setGifWeatherImage] = useState("");

  const { weather, temp, dt, dateCurrent } = props;

  useEffect(() => {
    const image = getWeatherImage(weather?.id, true);
    setGifWeatherImage(image);
  }, [weather]);

  if (!weather) {
    return <SkeletonWheaterCard {...props} />;
  }

  return (
    <>
      <S.WeatherCard {...props} data-testid="weather-card">
        <S.ContentTemp {...props}>
          <S.TemperatureCard {...props}>
            {temp?.toFixed(0)}
            <S.CelsiusText>Â°C</S.CelsiusText>
          </S.TemperatureCard>
          <S.DateCurrent {...props}>{dateCurrent}</S.DateCurrent>
        </S.ContentTemp>

        <S.IconCard {...props}>
          <Player autoplay loop src={gifWeatherImage} />
        </S.IconCard>
        <S.Description {...props}>{weather?.description}</S.Description>
        {dt && <S.Description {...props}>{dt}</S.Description>}
      </S.WeatherCard>
    </>
  );
};

export default React.memo(WeatherCard);
