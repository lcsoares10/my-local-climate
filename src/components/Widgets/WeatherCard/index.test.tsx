import { render, screen } from "@testing-library/react";
import Theme from "../../../theme";
import WeatherCard from "../WeatherCard";
import { weatherMock, tempMock, dateCurrentMock, dtMock } from "./mock";

describe("<TemperatureMaxMin />", () => {
  it("should view in display skeleton", async () => {
    render(
      <Theme>
        <WeatherCard weather={null} />
      </Theme>
    );

    expect(screen.getByTestId("skeleton-weather-card")).toBeTruthy();
  });

  it("should view in display inline ", async () => {
    const { container, debug } = render(
      <Theme>
        <WeatherCard
          weather={weatherMock}
          temp={tempMock}
          inline
          dateCurrent={dateCurrentMock}
        />
      </Theme>
    );

    expect(screen.getByText(`${weatherMock.description}`)).toBeInTheDocument();
    expect(screen.getByTestId("image-mock")).toBeInTheDocument();
    expect(screen.getByText(`${tempMock.toFixed(0)}`)).toBeInTheDocument();
    expect(screen.getByText(`°C`)).toBeInTheDocument();
    expect(screen.getByText(dateCurrentMock)).toBeInTheDocument();
    expect(screen.getByTestId("weather-card")).toHaveStyle({
      flexDirection: "row",
    });

    expect(screen.getByTestId("weather-card")).toMatchSnapshot();
  });

  it("should view in display card ", async () => {
    render(
      <Theme>
        <WeatherCard weather={weatherMock} temp={tempMock} dt={dtMock} />
      </Theme>
    );

    expect(screen.getByText(`${weatherMock.description}`)).toBeInTheDocument();
    expect(screen.getByTestId("image-mock")).toBeInTheDocument();
    expect(screen.getByText(`${tempMock.toFixed(0)}`)).toBeInTheDocument();
    expect(screen.getByText(`°C`)).toBeInTheDocument();
    expect(screen.getByText(dtMock)).toBeInTheDocument();

    expect(screen.getByTestId("weather-card")).toHaveStyle({
      flexDirection: "column",
    });
  });
});
