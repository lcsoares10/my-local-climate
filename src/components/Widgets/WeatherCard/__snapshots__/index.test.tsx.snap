// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TemperatureMaxMin /> should view in display inline  1`] = `
<div
  class="sc-gsDKAQ kQPEAc"
  data-testid="weather-card"
>
  <div
    class="sc-eCImPb gaXuTU"
  >
    <p
      class="sc-jRQBWg RIONF"
    >
      25
      <span
        class="sc-gKclnd iiOJEI"
      >
        Â°C
      </span>
    </p>
    <p
      class="sc-iCfMLu cJUUDS"
    >
      21/01/2022 15:03
    </p>
  </div>
  <div
    class="sc-hKwDye fRzwDB"
  >
    <div
      data-testid=" lottie-mock"
    >
      Mock Lottie
    </div>
  </div>
  <div
    class="sc-dkPtRN cBUYje"
  >
    Chuva leve
  </div>
</div>
`;
